query-engine/core/src/schema_builder/output_types/aggregation/group_by.rs

```diff
@@ -93,6 +93,8 @@ fn scalar_fields(ctx: &mut BuilderContext, model: &ModelRef) -> Vec<OutputField>
 
     fields
         .into_iter()
-        .map(|f| field(f.name.clone(), vec![], map_scalar_output_type_for_field(ctx, &f), None).optional())
+        .map(|f| {
+            field(f.name.clone(), vec![], map_scalar_output_type_for_field(ctx, &f), None).nullable_if(!f.is_required)
+        })
```

Comment:
actual fix - fields were already properly set as optional in the nested aggregate types. They weren't on top-level scalar fields though